drivers/bluetooth/hci/userchan.c

```diff
@@ -93,24 +93,70 @@ static bool is_hci_event_discardable(const uint8_t *evt_data)
 	}
 }
 
-static struct net_buf *get_rx(const uint8_t *buf)
+static struct net_buf *get_rx_evt(const uint8_t *data)
 {
-	bool discardable = false;
-	k_timeout_t timeout = K_FOREVER;
+	bool discardable = is_hci_event_discardable(data);
+	struct net_buf *buf;
+
+	do {
+		buf = bt_buf_get_evt(data[1], discardable, discardable ? K_NO_WAIT : K_SECONDS(10));
+		if (buf == NULL) {
+			if (discardable) {
+				LOG_DBG("Discardable buffer pool full, ignoring event");
+				return buf;
+			}
```

Comment:
Ditto, this will make it slightly different than IPC, but would it make sense to apply `LOG_DBG_RATELIMIT` here instead? Just in case that we get a lot of e.g. advertising reports that we want to discard

Reply:
ok, I'll see on how this 'rates' and updated if it is reasonable :)